### PR TITLE
Emulate iterat reas removal #84

### PR DESCRIPTION
**Pull request by irina-dragoste**
Pull request for issue #84.
This implementation sets QA correctness to INCORRECT for any KB change. The correctness can be further refined (ex: when there were no rules in the initial KB and only fact addition has been performed, when the same statements have been added and removed, etc)

### CODE REVIEW

File: vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/KnowledgeBase.java [hunk 1]
```diff
@@ -220,6 +256,74 @@ public void addStatements(Statement... statements) {
...
+	 * @return true, if the knowledge base has changed.
+	 */
+	boolean doRemoveStatement(Statement statement) {
+		Validate.notNull(statement, "Statement cannot be Null!");
+
+		if (this.statements.contains(statement) && statement.accept(this.removeStatementVisitor)) {
+			this.statements.remove(statement);
+			return true;
+		}
+		return false;
```
Comment (larryjgonzalez) [inline vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/KnowledgeBase.java]: Should we use `else { return false;  }` ?

File: vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/KnowledgeBase.java [hunk 2]
```diff
@@ -220,6 +256,74 @@ public void addStatements(Statement... statements) {
...
+	}
+
+	/**
+	 * Removes a collection of statements to the knowledge base.
+	 * 
+	 * @param statements
```
Comment (larryjgonzalez) [inline vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/KnowledgeBase.java]: Should we describe the parameter, instead of just name it?

**Comment by CerielJacobs** [comment]
I would be more concerned about the javadoc, which says that the statement is added, not removed :-)

Best,	Ceriel

File: vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/KnowledgeBase.java (hunk 1 above)
Comment (irina-dragoste) [reply vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/KnowledgeBase.java]: I don't think it's necessary.

File: vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/KnowledgeBase.java (hunk 2 above)
Comment (irina-dragoste) [reply vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/KnowledgeBase.java]: done

**Comment by irina-dragoste** [comment]
> I would be more concerned about the javadoc, which says that the statement is added, not removed :-) Best, Ceriel

Thank you for catching this,
done.

**Comment by larryjgonzalez** [comment]
* I've used the auto-formatter from eclipse in several files.
* I don't have suggestions for changes in classes under `src/main` directory
* I have several suggestions for the classes under `src/test` directory. But they should be addressed in #96

**Pull request merged**